app/services/noko_service.rb

```diff
@@ -23,29 +23,38 @@ def self.import_projects
 
   def self.import_entries(start_date, end_date)
     result = client.get_entries(from: start_date, to: end_date)
-    save_entries_for(result)
+    missing_projects = []
+    save_entries_for(result, missing_projects)
 
     if last = result.try(:link).try(:last)
       last_page = last.match(/page=(\d+)/)[1].to_i
       (2..last_page).each do |page|
         result = client.get_entries(from: start_date, to: end_date, page: page)
-        save_entries_for(result)
+        save_entries_for(result, missing_projects)
       end
     end
```

Comment:
Maybe we can do client.get_entries(from: start_date, to: end_date, per_page: 1000) so we don't have to do all this. I think 1000 entries is the maximum we can fetch per page, but i don't see that's a problem for us. What you think?

Reply:
@ercohen14 if you see we are `save_entries_for(result, missing_projects)` once for each result of a specified page (`result = client.get_entries(from: start_date, to: end_date, page: page)`) That is because the default entries per request is 30. so if we have 90 entries, this will be called 3 times. If we do `client.get_entries(from: start_date, to: end_date, per_page: 1000)` this will return the 90 entries in one call, since it will return a max of a 1000 entries for each call.

If instead

Reply:
I was thinking def `NokoService.import_entries` could be something like the following:
```ruby
def self.import_entries(start_date, end_date)
    result = client.get_entries(from: start_date, to: end_date, per_page: 1000)
    missing_projects = []
    save_entries_for(result, missing_projects)    

    if missing_projects.present?
      puts "The following time entries are missing a project. They must be updated before import: "
      missing_projects.each{ |missing| puts missing }
    end
end
```